Sources/AuthFoundation/Debugging/APIRequestObserver.swift

```diff
@@ -53,26 +53,46 @@ public final class DebugAPIRequestObserver: OAuth2ClientDelegate {
     }()
     
     /// Indicates if HTTP request and response headers should be logged.
-    public var showHeaders = false
-    
+    public var showHeaders: Bool {
+        get {
+            lock.withLock {
+                _showHeaders
+            }
+        }
+        set {
+            lock.withLock {
+                _showHeaders = newValue
+            }
+        }
+    }
+
     /// Convenience flag that automatically binds newly-created ``OAuth2Client`` instances to the debug observer.
-    public var observeAllOAuth2Clients: Bool = false {
-        didSet {
-            if observeAllOAuth2Clients {
-                oauth2Observer = NotificationCenter.default.addObserver(
-                    forName: .oauth2ClientCreated,
-                    object: nil,
-                    queue: nil,
-                    using: { [weak self] notification in
-                        guard let self = self,
-                              let client = notification.object as? OAuth2Client
-                        else {
-                            return
-                        }
-                        client.add(delegate: self)
-                    })
-            } else {
-                oauth2Observer = nil
+    nonisolated(unsafe) public var observeAllOAuth2Clients: Bool {
+        get {
+            lock.withLock {
+                _observeAllOAuth2Clients
+            }
+        }
+        set {
+            lock.withLock {
+                _observeAllOAuth2Clients = newValue
+
+                if _observeAllOAuth2Clients {
+                    oauth2Observer = TaskData.notificationCenter.addObserver(
+                        forName: .oauth2ClientCreated,
+                        object: nil,
+                        queue: nil,
+                        using: { [weak self] notification in
+                            guard let self = self,
+                                  let client = notification.object as? OAuth2Client
+                            else {
+                                return
+                            }
+                            client.add(delegate: self)
```

Comment:
Does this require synchronization? Or client cares about synchronization under the hood?

Reply:
This function is introduced by [the UsesDelegateCollection protocol](https://okta.github.io/okta-mobile-swift/development/authfoundation/documentation/authfoundation/usesdelegatecollection).  Since the `delegateCollection` property is a `let` value, and is `Sendable`, interacting with it from the OAuth2Client is considered safe.

I do notice that the DelegateCollection type itself is potentially susceptible to race conditions with mutations to its underlying `delegates` value, so I'll include an update that synchronizes concurrent access that mutates the list of delegates.